ndc-client/src/models.rs

```diff
@@ -156,10 +156,15 @@ pub enum Type {
 // ANCHOR: ComparisonOperatorDefinition
 /// The definition of a comparison operator on a scalar type
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[serde(tag = "type", rename_all = "snake_case")]
 #[schemars(title = "Comparison Operator Definition")]
-pub struct ComparisonOperatorDefinition {
-    /// The type of the argument to this operator
-    pub argument_type: Type,
+pub enum ComparisonOperatorDefinition {
+    Equal,
+    In,
+    Custom {
```

Comment:
Should we also include IsNull here?

Reply:
Yes I was considering that too. We can add that to the TODO list if the basic approach seems okay.

Reply:
Great - yeah the approach looks good to me!

Reply:
Actually, `IsNull` wouldn't be advertised for a particular scalar type - it applies to all columns that are nullable (maybe a capability).

But would we want other custom Unary comparison operators? (eg: IsPositive on an integer)

Reply:
Tracking what we discussed separately - let's keep `IsNull` as a special unary operator, and we can add custom unary operators if the need arises.